### PR TITLE
feat: Added remove

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: lib/primitives/object-element.js [hunk 1]
```diff
@@ -45,6 +45,14 @@ module.exports = function(BaseElement, ArrayElement, MemberElement) {
       );
     },
 
+    delete: function (name) {
+      this.content = this.content.filter(function (item) {
+        return item.key.toValue() !== name;
+      });
+
+      return this;
```
Comment (kylef) [inline lib/primitives/object-element.js]: Why would it return `this`? Why not return the deleted element like the other collection removal APIs in JS.

Comment (pksunkara) [reply lib/primitives/object-element.js]: Chaining. Check the tests. And I wanted to talk to you about this today.

File: lib/primitives/object-element.js [hunk 2]
```diff
@@ -45,6 +45,14 @@ module.exports = function(BaseElement, ArrayElement, MemberElement) {
       );
     },
 
+    delete: function (name) {
```
Comment (kylef) [inline lib/primitives/object-element.js]: `name`? Shouldn't this be key of the element?

File: README.md [hunk 3]
```diff
@@ -523,6 +523,18 @@ var objectElement = minim.toElement({ foo: 'bar' });
 var keys = objectElement.keys() // ['foo']
 ```
 
+##### delete
```
Comment (kylef) [inline README.md]: I think `remove` is a better name for this method as we're removing an element from the object. Deleting would imply the element is gone, but if I had a prior reference to it I can keep using the element. It is not "deleted".

File: README.md [hunk 4]
```diff
@@ -523,6 +523,18 @@ var objectElement = minim.toElement({ foo: 'bar' });
...
+
+The `delete` method deletes a key from the `ObjectElement` instance.
```
Comment (kylef) [inline README.md]: I think it's worth mentioning it's the key's content (value) and not the key (element) itself. This method does not accept the key only the keys value.

**Review by kylef — COMMENTED** [review]
Do you want to make this this method apply to arrays to so we can close off https://github.com/refractproject/minim/issues/59 too?

File: README.md (hunk 4 above)
Comment (pksunkara) [reply README.md]: What? I don't understand the above sentence

Comment (kylef) [reply README.md]: Both the following are not supported:

```js
object.delete(memberElement);
object.delete(keyElement);
```

Only this works:

```js
object.delete(keyValue);
```

Comment (pksunkara) [reply README.md]: Ah. Shouldn't the example give that away? All the other examples do the same thing. Let's not confuse the users by introducing an explanation on not supporting elements.

File: lib/primitives/object-element.js (hunk 2 above)
Comment (pksunkara) [reply lib/primitives/object-element.js]: It's the name of the key. This is the function signature being used everywhere. I didn't want to change it.

**Comment by pksunkara** [comment]
@kylef Updated.

File: lib/primitives/object-element.js (hunk 1 above)
Comment (kylef) [reply lib/primitives/object-element.js]: I don't think it makes sense to return the collection. Unless it was returning a new collection with the removed item (non mutating).

```javascript
const newObject = object.removing('somekey');
```

I think its far more likely you would want to chain the deleted element itself. This remove method doesn't offer the caller any feedback on whether the element was removed from the collection.

```javascript
if (object.remove('request')) {
    console.log('there was a request and we removed it');
}
```

```javascript
const response = object.remove('response');
if (response) {
    console.log('there was a response and we removed it');
    // do something with response.statusCode
}
```

Comment (kylef) [reply lib/primitives/object-element.js]: Can you please add a test case for removing an element that doesn't exist too.

**Comment by kylef** [comment]
@pksunkara can you rebase please.

**Comment by pksunkara** [comment]
Done.

**Review by kylef — APPROVED** [review]

**Pull request merged**